src/js/MarketData.jsx

```diff
@@ -0,0 +1,202 @@
+              duration={this.props.duration}
+              coin={this.props.coin}
+            />
+          ) : null}
+        </div>
+        <div className="row">
+          <div className="popup">
+            {this.props.hoverLoc ? (
+              <ToolTip
+                hoverLoc={this.props.hoverLoc}
+                activePoint={this.props.activePoint}
+              />
+            ) : null}
+          </div>
+        </div>
+        <div className="row">
+          <div className="chart">
+            {!this.props.data[coin][duration] ? <Spinner /> : null}
+            {!this.props.fetchingData ? (
+              <LineChart
+                duration={this.props.duration}
+                data={this.props.data}
+                coin={this.props.coin}
+                onChartHover={(a, b) => this.props.handleChartHover(a, b)}
+              />
+            ) : null}
+          </div>
+        </div>
+
+        <div className="row">
+          <select onChange={event => this.props.handleCoin(event, this.props)}>
+            <option value="Bitcoin">Bitcoin</option>
+            <option value="Ethereum">Ethereum</option>
+            <option value="Litecoin">Litecoin</option>
+          </select>;
+        </div>
+      </div>
+    );
+  }
+}
+
+const url = {
+  BTC: {
+    '1H |':
+      'https://min-api.cryptocompare.com/data/histominute?fsym=BTC&tsym=USD&limit=60&aggregate=1',
+    ' 1D |':
+      'https://min-api.cryptocompare.com/data/histohour?fsym=BTC&tsym=USD&limit=24&aggregate=1',
+    ' 1W |':
+      'https://min-api.cryptocompare.com/data/histohour?fsym=BTC&tsym=USD&limit=168',
+    ' 1M |': 'https://api.coindesk.com/v1/bpi/historical/close.json',
+    ' 1Y':
+      'https://min-api.cryptocompare.com/data/histoday?fsym=BTC&tsym=USD&limit=365&aggregate=1'
+  },
+  ETH: {
+    '1H |':
+      'https://min-api.cryptocompare.com/data/histominute?fsym=ETH&tsym=USD&limit=60&aggregate=1',
+    ' 1D |':
+      'https://min-api.cryptocompare.com/data/histohour?fsym=ETH&tsym=USD&limit=24&aggregate=1',
+    ' 1W |':
+      'https://min-api.cryptocompare.com/data/histohour?fsym=ETH&tsym=USD&limit=168',
+    ' 1M |':
+      'https://min-api.cryptocompare.com/data/histoday?fsym=ETH&tsym=USD&limit=30&aggregate=1',
+    ' 1Y':
+      'https://min-api.cryptocompare.com/data/histoday?fsym=ETH&tsym=USD&limit=365&aggregate=1'
+  },
+  LTC: {
+    '1H |':
+      'https://min-api.cryptocompare.com/data/histominute?fsym=LTC&tsym=USD&limit=60&aggregate=1',
+    ' 1D |':
+      'https://min-api.cryptocompare.com/data/histohour?fsym=LTC&tsym=USD&limit=24&aggregate=1',
+    ' 1W |':
+      'https://min-api.cryptocompare.com/data/histohour?fsym=LTC&tsym=USD&limit=168',
+    ' 1M |':
+      'https://min-api.cryptocompare.com/data/histoday?fsym=LTC&tsym=USD&limit=30&aggregate=1',
+    ' 1Y':
+      'https://min-api.cryptocompare.com/data/histoday?fsym=LTC&tsym=USD&limit=365&aggregate=1'
+  }
+};
+
+const mapStateToProps = state => ({
```

Comment:
Again, destructuring would be nice here

Reply:
can't because fetchingData != data.isFetching

Reply:
Sure but you could still write:
```js
const mapStateToProps = ({coin, duration, data, hoverLoc, activePoint }) => ({
  coin,
  duration,
  data,
  hoverLoc,
  activePoint,
  fetchingData: data.isFetching
});
```

But it's not important 😃